serialization/python/json/serialization/json_parse_node.py

```diff
@@ -0,0 +1,238 @@
+        """
+        return JsonParseNode(self._json_node[identifier])
+
+    def get_boolean_value(self) -> Optional[bool]:
+        """Gets the boolean value of the json node
+        Returns:
+            bool: The boolean value of the node
+        """
+        return bool(self._json_node)
+
+    def get_int_value(self) -> Optional[int]:
+        """Gets the integer value of the json node
+        Returns:
+            int: The integer value of the node
+        """
+        return int(self._json_node)
+
+    def get_float_value(self) -> Optional[float]:
+        """Gets the float value of the json node
+        Returns:
+            float: The integer value of the node
+        """
+        return float(self._json_node)
+
+    def get_uuid_value(self) -> Optional[UUID]:
+        """Gets the UUID value of the json node
+        Returns:
+            UUID: The GUID value of the node
+        """
+        return UUID(self._json_node)
+
+    def get_datetime_offset_value(self) -> Optional[datetime]:
+        """Gets the datetime offset value of the json node
+        Returns:
+            datetime: The datetime offset value of the node
+        """
+        datetime_str = str(self._json_node)
+        if datetime_str:
+            datetime_obj = parser.parse(datetime_str)
+            return datetime_obj
+        return None
+
+    def get_timedelta_value(self) -> Optional[timedelta]:
+        """Gets the timedelta value of the node
+        Returns:
+            timedelta: The timedelta value of the node
+        """
+        datetime_str = str(self._json_node)
+        if datetime_str:
+            datetime_obj = parser.parse(datetime_str)
+            return datetime.timedelta(hours=datetime_obj.hour,
+                                      minutes=datetime_obj.minute,
+                                      seconds=datetime_obj.second)
+        return None
+
+    def get_date_value(self) -> Optional[date]:
+        """Gets the date value of the node
+        Returns:
+            date: The datevalue of the node in terms on year, month, and day.
+        """
+        datetime_str = str(self._json_node)
+        if datetime_str:
+            datetime_obj = parser.parse(datetime_str)
+            return datetime_obj.date()
+        return None
+
+    def get_time_value(self) -> Optional[time]:
+        """Gets the time value of the node
+        Returns:
+            time: The time value of the node in terms of hour, minute, and second.
+        """
+        datetime_str = str(self._json_node)
+        if datetime_str:
+            datetime_obj = parser.parse(datetime_str)
+            return datetime_obj.time()
+        return None
+
+    def get_collection_of_primitive_values(self) -> Optional[List[T]]:
+        """Gets the collection of primitive values of the node
+        Returns:
+            List[T]: The collection of primitive values
+        """
+        def func(item):
```

Comment:
the behaviour should be driven by the generic type (what we want) and not the variable type (what it is). Failing if there's a mismatch is ok. It meant something went wrong